File: database/src/main/java/org/difft/app/database/models/GroupCryptoKeysModel.java
package org.difft.app.database.models;

import com.tencent.wcdb.WCDBField;
import com.tencent.wcdb.WCDBTableCoding;

import java.util.Objects;

@WCDBTableCoding
public class GroupCryptoKeysModel {
    @WCDBField(isPrimary = true)
    public String gid;

    @WCDBField
    public String rGroup;

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        GroupCryptoKeysModel that = (GroupCryptoKeysModel) o;
        return Objects.equals(gid, that.gid) && Objects.equals(rGroup, that.rGroup);
    }

    @Override
    public int hashCode() {
        return Objects.hash(gid, rGroup);
    }
}

File: database/src/main/java/org/difft/app/database/models/GroupCryptoKeysModel.kt
package org.difft.app.database.models

import com.tencent.wcdb.WCDBField
import com.tencent.wcdb.WCDBTableCoding

@WCDBTableCoding
class GroupCryptoKeysModel {
    @WCDBField(isPrimary = true)
    var gid: String? = null

    @WCDBField
    var rGroup: String? = null

    /**
     * Crypto-key generation. Baseline 0 = original/un-rotated (also what existing
     * rows read as: the column is auto-added by [db.createTable] and NULL reads as 0).
     * Server assigns monotonically increasing versions starting at 1 on each rotate.
     */
    @WCDBField
    var keyVersion: Int = 0

    override fun equals(other: Any?): Boolean {
        if (other == null || javaClass != other.javaClass) return false
        other as GroupCryptoKeysModel
        return gid == other.gid && rGroup == other.rGroup && keyVersion == other.keyVersion
    }

    override fun hashCode(): Int = java.util.Objects.hash(gid, rGroup, keyVersion)
}
